charmdir.go

```diff
@@ -174,9 +173,13 @@ func resolveSymlinkedRoot(rootPath string) (string, error) {
 // ArchiveTo creates a charm file from the charm expanded in dir.
 // By convention a charm archive should have a ".charm" suffix.
 func (dir *CharmDir) ArchiveTo(w io.Writer) error {
-	versionString, err := dir.MaybeGenerateVersionString()
+	versionString, vcsType, err := dir.MaybeGenerateVersionString()
 	if err != nil {
-		logger.Warningf("version string generation failed : %v", err)
+		// Just to be safe, ensure version is "" on error.
+		versionString = ""
+		logger.Warningf(
+			"%s version string generation failed : %v\nThis means that the charm version won't show in juju status.",
```

Comment:
I wonder if you want to have %q here instead... reading through https://golang.org/pkg/fmt/, %q seems better

Reply:
I don't mind %s as the message reads fine with it. Changed to %q though in case it helps.